datalad/distribution/publish.py

```diff
@@ -255,7 +256,10 @@ class Publish(Interface):
         recursion_limit=recursion_limit,
         git_opts=git_opts,
         annex_opts=annex_opts,
-        annex_copy_opts=annex_copy_opts
+        annex_copy_opts=annex_copy_opts,
+        # "interferes" with skip_failing which is actually "skip_missing"
+        # TODO: unify as --missing=(skip|inherit) ?
```

Comment:
so, semantically it is "skip missing" not "failing" since not that some publish of a sub-dataset has failed (which can happen) and we would like to skip it then.
So do you think I should tackle RF `--skip-failing` + `--inherit-settings` into `--missing=(fail|skip|inherit)` within this PR?  I guess I could as well do it